cmd/crc/cmd/daemon_windows.go

```diff
@@ -1,14 +1,86 @@
 package cmd
 
 import (
+	"fmt"
 	"os"
 
 	"github.com/code-ready/crc/pkg/crc/api"
 	"github.com/code-ready/crc/pkg/crc/constants"
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/debug"
+	"golang.org/x/sys/windows/svc/eventlog"
 )
 
 func runDaemon() {
 	// Remove if an old socket is present
 	os.Remove(constants.DaemonSocketPath)
-	api.RunDaemonService("CodeReady Containers", false, newConfig, newMachineWithConfig)
+	RunDaemonService("CodeReady Containers", false)
+}
+
+var elog debug.Log
+
+type daemonService struct {
+	socketPath string
+}
+
+func (m *daemonService) Execute(args []string, r <-chan svc.ChangeRequest, changes chan<- svc.Status) (ssec bool, errno uint32) {
+	const cmdsAccepted = svc.AcceptStop | svc.AcceptShutdown
+	crcAPIServer, err := api.CreateServer(m.socketPath, newConfig, newMachineWithConfig)
+	if err != nil {
+		elog.Error(1, fmt.Sprintf("Failed to start CRC daemon service: %v", err)) // nolint
+		return
+	}
+	// Windows servivce manager calls OnStart when starting the service
+	// and it calls this method, which the service manager expects to handle
+	// commands like stop, shutdown. crcAPIServer.Serve is being called as a go routine
+	// since we don't want it block
+	go crcAPIServer.Serve()
+	changes <- svc.Status{State: svc.Running, Accepts: cmdsAccepted}
+
+	MainLoop(r, changes)
+
+	// Above MainLoop will return when the daemon service is stopped
+	changes <- svc.Status{State: svc.Stopped}
+	return
+}
+
+func MainLoop(r <-chan svc.ChangeRequest, changes chan<- svc.Status) {
+	for c := range r {
+		switch c.Cmd {
+		case svc.Interrogate:
+			changes <- c.CurrentStatus
+		case svc.Stop, svc.Shutdown:
+			_ = elog.Info(1, "Shutting down CRC daemon service")
+			return
+		default:
+			_ = elog.Error(1, fmt.Sprintf("unexpected control request #%d", c))
+		}
+	}
+}
+
+func RunDaemonService(name string, isDebug bool) {
```

Comment:
The windows service for the daemon i was planning to remove because of the "Runs as admin issue.." in PR https://github.com/code-ready/crc/pull/1781

I guess we'll still need this for the `admin-helper`?

Reply:
I didnt see that you wanted to remove it :)

For the admin helper, yes we might need it!